.github/workflows/rebuild_native_libs.yml

```diff
@@ -0,0 +1,216 @@
+# It splits the building process into 3 build jobs:
+# 1. build-freebsd - Builds native libraries for x64 FreeBSD
+# 2. build-macos - Builds native libraries for MacOS: both ARM and x64.
+#    It uses native runners for each platform, because cross compilation on MacOS is complicated.
+# 3. build-others - Builds native libraries for x64 Linux, ARM Linux and Windows.
+#    It uses cross-compilation for ARM Linux and Windows.
+#
+# Each build job saves the resulting binaries to the cache under a unique key
+# When all build jobs are finished, the collect-commit-and-push job restores the binaries from the cache
+# and pushes them to the current branch.
+
+jobs:
+  build-freebsd:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build in FreeBSD
+        # @d44bf830e7d is the same as the tag v1, but it's guaranteed to be immutable.
+        uses: vmactions/freebsd-vm@d44bf830e7d850111d27aa120d468c905e5980f4
+        with:
+          # FreeBSD 13.2
+          release: 13.2
+          usesh: true
+          prepare: |
+            pkg install -y curl cmake gcc git nasm openjdk17
+          run: |
+            export JAVA_HOME=/usr/local/openjdk17
+            cd core
+            cmake -B build/release -DCMAKE_BUILD_TYPE=Release
+            cmake --build build/release --config Release
+      - name: log git status
+        run: |
+          git status
+      - name: Save FreeBSD binary to Cache
+        uses: actions/cache/save@v3
+        with:
+          path: |
+            core/src/main/resources/io/questdb/bin/freebsd/libquestdb.so
+          key: nativelibs-freebsd-${{ github.sha }}
+  build-macos:
+    strategy:
+      matrix:
+        # macos-12-xlarge = ARM M1
+        # macos-latest = x64
+        # if you change OS definitions then you need to change conditions in cache-save steps below
+        os: [ macos-13-xlarge, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install tooling
+        run: |
+          brew install make cmake gcc nasm
+      - name: Build MacOS
+        run: |
+          cd core
+          cmake -B build/release -DCMAKE_BUILD_TYPE=Release
+          cmake --build build/release --config Release
+          cd ..
+      - name: Save ARM MacOS binary to Cache
+        if: ${{ matrix.os == 'macos-13-xlarge' }}
+        uses: actions/cache/save@v3
+        with:
+          path: |
+            core/src/main/resources/io/questdb/bin/armosx/libquestdb.dylib
+          key: nativelibs-armosx-${{ github.sha }}
+      - name: Save x64 MacOS binary to Cache
+        if: ${{ matrix.os == 'macos-latest' }}
+        uses: actions/cache/save@v3
+        with:
+          path: |
+            core/src/main/resources/io/questdb/bin/osx/libquestdb.dylib
+          key: nativelibs-osx-${{ github.sha }}
+  build-others:
+    runs-on: ubuntu-latest
```

Comment:
Which glibc min version shall we have here? If I'm not mistaken, currently we use Ubuntu 20.04 to make sure that we're compatible with glibc 2.29.

Reply:
good question! Azure Pipelines uses ubuntu-20.04, let me try the same here.

Reply:
testing here: https://github.com/jerrinot/questdb/actions/runs/6875301619

Reply:
the testing build passed 🎉

Reply:
nit: I'd add a comment on why we use this particular version, so that no one changes it accidentally.